[docs] Update Contacts calendar formats (#17956)
expo/expo: docs/pages/versions/unversioned/sdk/contacts.md

| department              | `string`                  | Job department.                                                                                                  | ✅   | ✅      |
| note                    | `string`                  | Additional information.                                                                                          | ✅\* | ✅      |
| imageAvailable          | `boolean`                 | Used for efficient retrieval of images.                                                                          | ✅   | ✅      |
| image                   | `Image`                   | Thumbnail image (ios: 320x320)                                                                                   | ✅   | ✅      |
| rawImage                | `Image`                   | Raw image without cropping, usually large.                                                                       | ✅   | ✅      |
| contactType             | `ContactType`             | Denoting a person or company.                                                                                    | ✅   | ✅      |

> | birthday                | `Date`                    | Birthday information in JS format.                                                                               | ✅   | ✅      |

Reviewer: ```suggestion
| birthday                | `Date`                    | Gregorian birthday information.                                                                               | ✅   | ✅      |
```

> | dates                   | `Date[]`                  | A list of other relevant user dates in JS format.                                                                | ✅   | ✅      |

Reviewer: ```suggestion
| dates                   | `Date[]`                  | A list of other labeled Gregorian dates.                                                                | ✅   | ✅      |
```

> | dates                   | `Date[]`                  | A list of other relevant user dates in Gregorian format.                                                         | ✅   | ✅      |

Reviewer: ```suggestion
| dates                   | `Date[]`                  | A labeled list of other relevant user dates in Gregorian format.                                                         | ✅   | ✅      |
```
